.circleci/config/commands/go_test.yml

```diff
@@ -58,21 +58,38 @@ steps:
         USE_DOCKER=1
         <</ parameters.use_docker >>
 
-        # Split Go tests by prior test times.  If use_docker is true, only run
-        # tests that depend on docker, otherwise only those that don't.
-        if [ $USE_DOCKER == 1 ]; then
-          package_names=$(go list -test -json ./... |
-            jq -r 'select(.Deps != null) |
-              select(any(.Deps[] ; contains("github.com/hashicorp/vault/helper/testhelpers/docker"))) |
-              .ForTest | select(. != null)' |
-              sort -u | grep -v vault/integ | circleci tests split --split-by=timings --timings-type=classname)
-        else
-          package_names=$(go list -test -json ./... |
-            jq -r 'select(.Deps != null) |
-              select(all(.Deps[] ; contains("github.com/hashicorp/vault/helper/testhelpers/docker")|not)) |
-              .ForTest | select(. != null)' |
-              sort -u | grep -v vault/integ | circleci tests split --split-by=timings --timings-type=classname)
-        fi
+        # Check all directories with a go.mod file
+        modules=('.' 'api' 'sdk')
+        all_package_names=""
+
+        for dir in "${modules[@]}"
+        do
+          cd $dir
+          # Split Go tests by prior test times.  If use_docker is true, only run
+          # tests that depend on docker, otherwise only those that don't.
+          # The appended true condition ensures the command will succeed if no packages are found
+          if [ $USE_DOCKER == 1 ]; then
+            package_names=$(go list -test -json ./... |
+              jq -r 'select(.Deps != null) |
+                select(any(.Deps[] ; contains("github.com/hashicorp/vault/helper/testhelpers/docker"))) |
+                .ForTest | select(. != null)' |
+                sort -u | grep -v vault/integ | circleci tests split --split-by=timings --timings-type=classname) || true
+          else
+            package_names=$(go list -test -json ./... |
+              jq -r 'select(.Deps != null) |
+                select(all(.Deps[] ; contains("github.com/hashicorp/vault/helper/testhelpers/docker")|not)) |
+                .ForTest | select(. != null)' |
+                sort -u | grep -v vault/integ | circleci tests split --split-by=timings --timings-type=classname) || true
+          fi
+          # Move back into root directory if needed
```

Comment:
Is this block necessary? Can we test it to see if we do indeed need to move backward? Also, is it ok to always assume we traverse back only one directory?

Reply:
The move backwards is necessary, cause we cd-d into a  sub-directory we ran `go list` on i.e we cd into `api`, get the package names, then cd back into root, before we cd into `sdk`.
About the assumption, that is a solid point, this does indeed assume we're only 1 step away from the root, which works currently for api and sdk, but may indeed break in the future. Maybe saving the `pwd` value before entering the for loop, and cd-ing into that here, makes sense ?

Reply:
Ended up changing this to use pushd/popd as josh suggested
bf0407d